--- a/contrib/svt-av1/module.defs	2023-12-12 19:07:02.563002053 -0500
+++ b/contrib/svt-av1/module.defs	2023-12-12 19:07:36.675069920 -0500
@@ -15,6 +15,7 @@
 SVT-AV1.CONFIGURE.shared      = -DBUILD_SHARED_LIBS=OFF
 SVT-AV1.CONFIGURE.extra       = -DBUILD_DEC=OFF -DBUILD_APPS=OFF -DREPRODUCIBLE_BUILDS=1
 SVT-AV1.CONFIGURE.extra      += -DCMAKE_INSTALL_LIBDIR=lib
+SVT-AV1.CONFIGURE.extra      += -DENABLE_AVX512=ON
 
 ifeq ($(GCC.O),$(filter $(GCC.O),size size-aggressive))
     SVT-AV1.CONFIGURE.extra += -DCMAKE_BUILD_TYPE=MinSizeRel
