--- a/media_driver/cmake/linux/media_compile_flags_linux.cmake	2023-11-10 16:00:32.182289902 -0500
+++ b/media_driver/cmake/linux/media_compile_flags_linux.cmake	2023-11-10 16:01:23.034436228 -0500
@@ -51,8 +51,6 @@
     -ffunction-sections
     -Wl,--gc-sections
 
-    # Enable c++14 features
-    -std=c++14
     # -m32 or -m64
     -m${ARCH}
 
@@ -110,6 +108,8 @@
     -fexceptions
     -fpermissive
     -fcheck-new
+    # Enable c++14 features
+    -std=c++14
 )
 
 if(NOT ${PLATFORM} STREQUAL "android")
@@ -112,13 +112,6 @@
     -fcheck-new
 )
 
-if(NOT ${PLATFORM} STREQUAL "android")
-    set(MEDIA_COMPILER_CXX_FLAGS_COMMON
-        ${MEDIA_COMPILER_CXX_FLAGS_COMMON}
-        -std=c++1y
-    )
-endif()
-
 set(MEDIA_COMPILER_FLAGS_RELEASE "")
 
 if(${UFO_VARIANT} STREQUAL "default")
--- a/Tools/bldsys/include/utils.cmake	2024-08-26 21:05:30.979553316 -0400
+++ b/Tools/bldsys/include/utils.cmake	2024-08-26 21:05:41.891712443 -0400
@@ -55,7 +55,7 @@
     endif (${PLATFORM} STREQUAL "ghs")
 
     if(NOT ${PLATFORM} STREQUAL "qnx")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
     endif()
 
     if (${PLATFORM} STREQUAL linux)
--- a/cmrtlib/linux/CMakeLists.txt	2024-08-26 20:27:26.697618847 -0400
+++ b/cmrtlib/linux/CMakeLists.txt	2024-06-10 11:47:55.000000000 -0400
@@ -28,7 +28,7 @@
 if (LATEST_CPP_NEEDED)
     set(CPP_STANDARD_OPTION -std=c++17)
 else()
-    set(CPP_STANDARD_OPTION -std=c++11)
+    set(CPP_STANDARD_OPTION -std=c++14)
 endif()
 
 # Set up compile options that will be used for the Linux build
